internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -178,24 +179,27 @@ func (r *Reconciler) reconcileNode(ctx context.Context, s *scope) (ctrl.Result,
 
 // getManagedLabels gets a map[string]string and returns another map[string]string
 // filtering out labels not managed by CAPI.
-func getManagedLabels(labels map[string]string) map[string]string {
+func (r *Reconciler) getManagedLabels(labels map[string]string) map[string]string {
 	managedLabels := make(map[string]string)
 	for key, value := range labels {
 		dnsSubdomainOrName := strings.Split(key, "/")[0]
-		if dnsSubdomainOrName == clusterv1.NodeRoleLabelPrefix {
-			managedLabels[key] = value
-		}
-		if dnsSubdomainOrName == clusterv1.NodeRestrictionLabelDomain || strings.HasSuffix(dnsSubdomainOrName, "."+clusterv1.NodeRestrictionLabelDomain) {
-			managedLabels[key] = value
-		}
-		if dnsSubdomainOrName == clusterv1.ManagedNodeLabelDomain || strings.HasSuffix(dnsSubdomainOrName, "."+clusterv1.ManagedNodeLabelDomain) {
+		if r.allowedDomainSync(dnsSubdomainOrName) {
 			managedLabels[key] = value
 		}
 	}
-
 	return managedLabels
 }
 
+func (r *Reconciler) allowedDomainSync(domain string) bool {
+	for _, v := range r.SyncMachineLabels {
+		matched, _ := filepath.Match(v, domain)
```

Comment:
What is the reason we're using filepath.Match that as per the documentation "matches shell file name patterns" to select labels to sync?

I would have expected something like regexp as it should be more intuitive to match regexp against the full labels keys instead of using file patch matching on a substring of the label key

Reply:
Switched to regex to make it more generic. Now it can be used for any kinds of labels (with or without a domain.)